### ISSUE TITLE
Upgrade error 0.35

### ISSUE THREAD

**Issue by usag1r** [twentyhq/twenty#9254]
After running `yarn command:prod upgrade-0.35` I got an error as below and Workspace is not loading anymore.
Other deployments upgraded successfully but a specific one threw this error and now not working.

[Nest] 598  - 12/26/2024, 7:29:23 PM   ERROR [WorkspaceSyncMetadataService] Sync of standard objects failed with:
[Nest] 598  - 12/26/2024, 7:29:23 PM   ERROR [WorkspaceSyncMetadataService] QueryFailedError: column "viewId" of relation "viewField" contains null values

**Comment by usag1r** [comment]
I was able to solve this by connecting to the database and deleting 5 views which didn't have viewId values from the database. It was tricky for me and I have no idea where they come from or why db migration didn't handle this.

After connecting to the postgres db:
`SET search_path TO workspace_xyz12319203abcasdklfasdlf;` (your unique workspace name should be replaced)
and I was able to list viewFields as following: 
`SELECT * FROM workspace_4l3fd7gu8x6fmdved757jk413.viewField;`

 then I deleted the ones with NULL for viewId and repeated `yarn command:prod upgrade-0.35`

**Comment by gboxxx** [comment]
I was having same issue and Claude helped me get to deleting NULL value viewFields. 

I'm still getting `[Nest] 711  - 12/29/2024, 8:51:12 PM   ERROR [WorkspaceSyncMetadataService] Sync of standard objects failed with:
[Nest] 711  - 12/29/2024, 8:51:12 PM   ERROR [WorkspaceSyncMetadataService] SyntaxError: Unexpected token ' in JSON at position 0` however.

Any advice?

**Comment by FelixMalfait** [comment]
Hey, the upgrade process isn't great yet sorry, they have to be incremental for now (we'll improve this). I think what probably happened here is that you skipped a version. In the 033 script there is something to cleanup the views

<img width="347" alt="Image" src="https://github.com/user-attachments/assets/33795833-642e-4c0b-bb49-afb5249bd044" />


@gboxxx would you have the full stack trace for this error? 
I would say the same thing, make sure to upgrade step by step and not skip version 

cc @ehconitin fyi

**Comment by usag1r** [comment]
Thank you @FelixMalfait , Happy New Year!

I would have a comment on the versioning and upgrade methodology advice in this case. Rather than advising to upgrade to each second decimal version, it's much more intuitive to include crucial upgrades only in first decimal versions such as 0.4, 0.5, 0.6 because Upgrade Documentation gives the impression that what matters is upgrading to 0.30, 0.40, 0.50 and so on...

Additionally such scripts should be explained and or provided as independent hot fixes (and their significance should be communicated properly) for people who are in trouble with upgrades. Otherwise, if you have multiple twenty CRM installations and take a couple months off from upgrades, you will potentially face with 50+ upgrades which isn't going to be practical.

**Comment by FelixMalfait** [comment]
@usag1r yes agree with your feedback. That will take a few months to improve but I think that's something where we should do much better by end of Q1!
